### PR TITLE
Validate change category during parsing

### PR DESCRIPTION
**Pull request by Gudahtt**
The change category is now validated as part of `parseChangelog`. Previously it was still being validated, but not until later on when the change categories were added to a `Changelog` instance. Validating it earlier in `parseChangelog` makes more sense now that we have types.

### CODE REVIEW

**Review by ryanml — APPROVED** [review]
++

File: src/parseChangelog.test.ts [hunk 1]
```diff
@@ -568,6 +568,32 @@ describe('parseChangelog', () => {
     ).toThrow(`Malformed category header: '### Ch-Ch-Ch-Ch-Changes'`);
```
Comment (rekmarks) [inline src/parseChangelog.test.ts:568]: > Ch-Ch-Ch-Ch-Changes

😂 💯 👨‍🎤

File: src/parseChangelog.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+  orderedChangeCategories,
+  unreleased,
+} from './constants';
 
 function truncated(line: string) {
   return line.length > 80 ? `${line.slice(0, 80)}...` : line;
 }
 
+function isValidChangeCategory(category: string): category is ChangeCategory {
+  return orderedChangeCategories.includes(category as ChangeCategory);
```
Comment (rekmarks) [inline src/parseChangelog.ts:13]: I think we could just index into `ChangeCategory` since the values and keys are identical, but hardly worth blocking on.

**Review by rekmarks — DISMISSED** [review]
LGTM! Just one nit you can take or leave.

File: src/parseChangelog.ts (hunk 2 above)
Comment (Gudahtt) [reply src/parseChangelog.ts]: Great idea - done in https://github.com/MetaMask/auto-changelog/pull/62/commits/d903e1def68ee4bbb4a9af8d33e2c03920a71059

**Review by rekmarks — APPROVED** [review]
🚀

**Pull request merged**